lib/stack_master/stack_definition.rb

```diff
@@ -7,15 +7,29 @@ class StackDefinition
       attribute :stack_name, String
       attribute :template, String
       attribute :tags, Hash
+      attribute :s3, Hash
       attribute :notification_arns, Array[String]
       attribute :base_dir, String
       attribute :secret_file, String
+      attribute :files, Array[String]
       attribute :stack_policy_file, String
       attribute :additional_parameter_lookup_dirs, Array[String]
     end
 
+    def template_dir
+      File.join(base_dir, 'templates')
+    end
+
     def template_file_path
-      File.join(base_dir, 'templates', template)
+      File.join(template_dir, template)
+    end
+
+    def files_to_upload
+      require 'pry'
+      binding.pry
```

Comment:
Sorry about the debug flags, I'll remove them.  I think you're right that it's not called. And its incorrect at this point anyway, I had to change how we sourced the files to upload since I wanted to preserve the full path of the files. just having the filename isn't enough because you need the relative path it should be uploaded to. 99% sure it just needs the files_to_upload in apply rather than this one.